### DISCUSSION TITLE
course/starbook/learn-word-rand

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#136] [General]
# course/starbook/learn-word-rand

使用GoFrame框架设计随机获取单词接口。提供路径words/rand的API，支持在1到300之间限制获取单词个数。使用OrderRandom实现随机查询，Limit限制查询返回数量。控制器调用逻辑处理查询结果，并进行接口测试以验证功能。

https://goframe.org/course/starbook/learn-word-rand

**Comment by Abner3721** [comment]
internal/controller/words/words_v1_rand_list.go

nternal/logic/words/learn_words.go

这两个文件里面的内容反了。

**Comment by chaogenb110** [comment]
为啥c可以当service在使用呢？

**Reply by wrm244** [reply]
建议先去看 go 基础，这个是接受者的方法调用，go 弱化了面向对象，详细见：https://golang.halfiisland.com/essential/base/80.method.html
